### PR TITLE
Add seed handler and tests

### PR DESCRIPTION
**Pull request by fritzo**
Blocked by https://github.com/pyro-ppl/pyro/pull/2061

## Tested

- added unit tests

### CODE REVIEW

File: test/test_minipyro.py [hunk 1]
```diff
@@ -90,6 +90,59 @@ def model(data=None):
...
+        with handlers.seed(rng_seed=0):
+            model()
+            with handlers.seed(rng_seed=0):
+                model()
+            actual = model()
+        assert_close(actual, expected)
+
+
+@pytest.mark.parametrize("backend", ["pyro", "minipyro", "funsor"])
+def test_rng_state(backend):
```
Comment (neerajprad) [inline test/test_minipyro.py]: Is the rng_state being tested because of the default seed handler in `pyro_backend`?

Comment (fritzo) [reply test/test_minipyro.py]: No, the rng_state is being tested irrelevant of the behavior of pyro_backend. I just wanted to add a test, and the natural place for that was in test_minipyro.py.

File: test/test_minipyro.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import pytest
 import torch
 from pyro.generic import distributions as dist
-from pyro.generic import infer, optim, pyro, pyro_backend
+from pyro.generic import handlers, infer, optim, pyro, pyro_backend
```
Comment (neerajprad) [inline test/test_minipyro.py]: You can probably start importing this from pyroapi now (since it is a test dependency)?

File: test/test_minipyro.py (hunk 1 above)
Comment (neerajprad) [reply test/test_minipyro.py]: It seems like there's a name clash with the test above?

Comment (fritzo) [reply test/test_minipyro.py]: fixed.

File: test/test_minipyro.py (hunk 2 above)
Comment (fritzo) [reply test/test_minipyro.py]: fixed.
